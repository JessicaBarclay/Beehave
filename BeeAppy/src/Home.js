import React from 'react';
import BeeButton from './BeeButton';
import { Text, View } from 'react-native';

export default class Home extends React.Component {
  render() {
    return(
      <View>
        <Text>Hi, welcome to BeeAppy </Text>
        <BeeButton />
      </View>
    );
  }
}
