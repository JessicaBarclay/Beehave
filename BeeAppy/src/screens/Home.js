import React from 'react';
import BeeButton from '../components/BeeButton';
import BeeFactButton from '../components/BeeFactButton';
import Donate from './Donate';
import style from '../styles/style.js';
import {
  StyleSheet,
  Text,
  View,
  Button,
  TouchableHighlight,
  Image
} from 'react-native';
import { StackNavigator } from 'react-navigation';

export default class Home extends React.Component {
  static navigationOptions = {
    title: 'BeeAppy',
  };
  render() {
    const { navigate } = this.props.navigation;
    return(
      <View>
        <Text>Hi, welcome to BeeAppy </Text>
        <Text style={style.homeTextStyle}>{'\n'}Spotted a bee?  Click the button!{'\n'}</Text>
        <BeeButton />
        <TouchableHighlight
          onPress={() => navigate('Donate')}>
          <Image
            style={style.imageButtonStyle}
            source={require('../img/honey.png')}
          />
        </TouchableHighlight>
        <TouchableHighlight
          onPress={() => navigate('BeeHive')}>
          <Image
            style={style.imageButtonStyle}
            source={require('../img/earth-globe.png')}
          />
        </TouchableHighlight>
        <BeeFactButton />
      </View>
    );
  }
}
