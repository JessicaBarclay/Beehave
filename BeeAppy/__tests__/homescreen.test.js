import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Home from '../src/Home';
import renderer from 'react-test-renderer';

describe('Home component', function() {
  it('renders a home page', () => {
    const homepage = renderer.create(<Home/>).toJSON();
    expect(homepage).toMatchSnapshot();
  });

  it('should render', function() {
    expect(render(<Home/ >).text()).toEqual('Hi');
  });
});
