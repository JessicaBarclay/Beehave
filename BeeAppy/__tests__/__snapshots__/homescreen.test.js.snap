// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders a home page 1`] = `
<View>
  Hi
</View>
`;
